### PR TITLE
[EGE-485] feat(st-select): new design for the search of elements

### PR DESCRIPTION
**Pull request by jdiaz-stratio**
### Documentation
- [x] Add the issue in PR description
- [x] Have changelog updated
- [x] You fill the milestone value
- [x] You add some label
- [x] Have example running in demo app
- [ ] Review id on demo is the same of egeo folder

### Code
- [x] Pass Sass lint task
- [ ] Have qaTags

### Tests
- [x] Have at least 70% tests coverage

### CODE REVIEW

File: src/lib/st-select/st-select.spec.ts [hunk 1]
```diff
@@ -565,31 +566,6 @@ describe('StSelectComponent', () => {
          fixture.destroy();
       });
 
-      it('Should filter list on search', () => {
```
Comment (emgaitan-stratio) [inline src/lib/st-select/st-select.spec.ts]: Fix test

Comment (jdiaz-stratio) [reply src/lib/st-select/st-select.spec.ts]: Done

File: src/lib/st-select/st-select.ts [hunk 2]
```diff
@@ -156,18 +162,25 @@ export class StSelectComponent implements AfterViewInit, ControlValueAccessor, O
     */
 
    getOptions(): void {
-      if (this.selected) {
-         this.onSearch(this.selected.label);
+      console.log('getOptions', this.searchInput.value);
```
Comment (emgaitan-stratio) [inline src/lib/st-select/st-select.ts]: Remove console log

File: src/lib/st-select/st-select.ts [hunk 3]
```diff
@@ -156,18 +162,25 @@ export class StSelectComponent implements AfterViewInit, ControlValueAccessor, O
...
+      if (this.searchInput.value) {
+         this.onSearch(this.searchInput.value);
       }
    }
 
    onSearch(value: string): void {
+      console.log('onSearch');
```
Comment (emgaitan-stratio) [inline src/lib/st-select/st-select.ts]: Remove console log

File: src/lib/st-select/st-select.ts [hunk 4]
```diff
@@ -206,13 +219,17 @@ export class StSelectComponent implements AfterViewInit, ControlValueAccessor, O
    ngOnInit(): void {
       if (this.search) {
          this._searchInputSubscription = this.searchInput.valueChanges.subscribe(this.onSearch.bind(this));
+         this._subscriptionSearchClearButton = this.searchInput.valueChanges.subscribe((val) => this.showClear = (val && val.length > 0));
       }
    }
 
    ngOnDestroy(): void {
       if (this._searchInputSubscription) {
```
Comment (emgaitan-stratio) [inline src/lib/st-select/st-select.ts]: You can use the same subscription array

File: src/lib/st-select/st-select.ts (hunk 3 above)
Comment (jdiaz-stratio) [reply src/lib/st-select/st-select.ts]: Done

File: src/lib/st-select/st-select.ts (hunk 4 above)
Comment (jdiaz-stratio) [reply src/lib/st-select/st-select.ts]: Done

File: src/lib/st-select/st-select.ts (hunk 2 above)
Comment (jdiaz-stratio) [reply src/lib/st-select/st-select.ts]: Done

File: src/lib/st-select/st-select.ts [hunk 5]
```diff
@@ -206,6 +216,8 @@ export class StSelectComponent implements AfterViewInit, ControlValueAccessor, O
    ngOnInit(): void {
       if (this.search) {
          this._searchInputSubscription = this.searchInput.valueChanges.subscribe(this.onSearch.bind(this));
```
Comment (emgaitan-stratio) [inline src/lib/st-select/st-select.ts]: Use only one "subscribe" function

Comment (jdiaz-stratio) [reply src/lib/st-select/st-select.ts]: Done

**Pull request merged**